services/core-web/common/models/documents/document.ts

```diff
@@ -154,24 +123,74 @@ export class MineDocument {
 
 export class MajorMineApplicationDocument extends MineDocument {
 
+  public project_summary_document_xref: { project_summary_document_type_code: string };
+  public project_decision_package_document_xref: { project_decision_package_document_type_code: string };
+  public information_requirements_table_document_xref: { information_requirements_table_document_type_code: string };
+  public major_mine_application_document_xref: { major_mine_application_document_type_code: string };
+
   public major_mine_application_document_type_code: string;
   public versions: MajorMineApplicationDocument[];
 
   constructor(jsonObject: any) {
     super(jsonObject);
     this.major_mine_application_document_type_code =
       jsonObject.major_mine_application_document_type_code;
+    this.category_code = this.determineCategoryCode(jsonObject);
+    this.setCalculatedProperties(jsonObject);
   }
 
-  protected makeChild(params: any, constructorArgs: any) {
+  protected determineCategoryCode(jsonObject: any): string | undefined {
+    const {
+      project_summary_document_xref,
+      project_decision_package_document_xref,
+      information_requirements_table_document_xref,
+      major_mine_application_document_xref,
+    } = jsonObject;
 
+    return (
+      project_summary_document_xref?.project_summary_document_type_code
+      ?? project_decision_package_document_xref?.project_decision_package_document_type_code
+      ?? information_requirements_table_document_xref?.information_requirements_table_document_type_code
+      ?? major_mine_application_document_xref?.major_mine_application_document_type_code
+    );
+  }
+
+  protected makeChild(params: any, constructorArgs: any) {
     return new MajorMineApplicationDocument({
       ...params,
       major_mine_application_document_type_code:
         constructorArgs.major_mine_application_document_type_code,
     });
   }
 
+  protected setCalculatedProperties(jsonObject: any) {
```

Comment:
I don't think the content of this needs to be duplicated from the base model, since the base model passes the jsonObject to the makeChild method.